examples/architectureComponentsExample/src/main/java/io/realm/examples/arch/livemodel/LiveRealmObject.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.arch.livemodel;
+
+import android.arch.lifecycle.LiveData;
+import android.support.annotation.MainThread;
+
+import io.realm.ObjectChangeSet;
+import io.realm.RealmModel;
+import io.realm.RealmObject;
+import io.realm.RealmObjectChangeListener;
+
+
+public class LiveRealmObject<T extends RealmModel> extends LiveData<T> {
```

Comment:
Yup, there are many ways you could split the responsibility. I have only played around with ViewModel/LiveData in toy examples so far and at least based on that I could see arguments for controlling the listeners in either the ViewModel or the LiveData. Neither seems inherently wrong to me.

One argument that favors LiveData is from Yirgits talk: The LocationLiveData class encapsulate the listeners in the LiveData class: https://youtu.be/bEKNi1JOrNs?t=14m43s